### PR TITLE
fix: update code coverage script and prevent similar problems in the future

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Fix errors in code coverage script.
- Run a small version of the code coverage script in PRs when they're out of draft.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by hanabi1224 — APPROVED** [review]

**Comment by lemmih** [comment]
While I appreciate the quick review, there is more work to do in this PR before it is ready. :)

File: scripts/gen_coverage_report.sh [hunk 1]
```diff
@@ -28,37 +30,42 @@ function cov {
...
+cov forest-tool snapshot fetch --chain calibnet --directory "$TMP_DIR"
+SNAPSHOT_PATH=$(find "$TMP_DIR" -name \*.forest.car.zst | head -n 1)
 cov forest --chain calibnet --encrypt-keystore false --import-snapshot "$SNAPSHOT_PATH" --height=-200 --detach --track-peak-rss --save-token "$TOKEN_PATH"
 cov forest-cli sync wait
 cov forest-cli sync status
-cov forest-cli --chain calibnet db gc
 cov forest-tool db stats --chain calibnet
-cov forest-cli snapshot export
-cov forest-cli snapshot export
+# cov forest-cli snapshot export
```
Comment (LesnyRumcajs) [inline scripts/gen_coverage_report.sh:51]: do we want it or not?

**Review by LesnyRumcajs — APPROVED** [review]

File: scripts/gen_coverage_report.sh (hunk 1 above)
Comment (lemmih) [reply scripts/gen_coverage_report.sh:51]: Oops, we do want it.

**Comment by lemmih** [comment]
@LesnyRumcajs Need a re-review before this can be merged.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**